inst/ad_advs.R

```diff
@@ -0,0 +1,260 @@
+# Name: ADVS
+#
+# Label: Vital Signs Analysis dataset
+#
+# Input: adsl, vs
+
+
+# Attach/load required packages ----
+library(admiral)
+library(admiralmetabolic)
+library(tibble)
+library(dplyr)
+library(stringr)
+
+
+# Define look-up tables ----
+# See function documentation for `derive_vars_merged_lookup()` and
+# `derive_vars_cat()` for information of how look-up tables can be used:
+# (https://pharmaverse.github.io/admiral/reference/derive_vars_merged_lookup.html)
+# (https://pharmaverse.github.io/admiral/reference/derive_vars_cat.html)
+
+# Parameter look-up
+param_lookup <- tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN, ~PARCAT1, ~PARCAT1N,
+  "HEIGHT", "HEIGHT", "Height (cm)", 1, "Subject Characteristic", 1,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 2, "Subject Characteristic", 1,
+  "BMI", "BMI", "Body Mass Index(kg/m^2)", 3, "Subject Characteristic", 1,
+  "HIPCIR", "HIPCIR", "Hip Circumference (cm)", 4, "Subject Characteristic", 1,
+  "WSTCIR", "WSTCIR", "Waist Circumference (cm)", 5, "Subject Characteristic", 1,
+  "DIABP", "DIABP", "Diastolic Blood Pressure (mmHg)", 6, "Vital Sign", 2,
+  "PULSE", "PULSE", "Pulse Rate (beats/min)", 7, "Vital Sign", 2,
+  "SYSBP", "SYSBP", "Systolic Blood Pressure (mmHg)", 8, "Vital Sign", 2,
+  "TEMP", "TEMP", "Temperature (C)", 9, "Vital Sign", 2
+)
+
+# Analysis categories look-up (conditional look-up)
+avalcat_lookup <- exprs(
+  ~PARAMCD, ~condition,                ~AVALCAT1,           ~AVALCA1N,
+  "BMI",    AVAL < 18.5,               "Underweight",       1,
+  "BMI",    AVAL >= 18.5 & AVAL < 25,  "Normal weight",     2,
+  "BMI",    AVAL >= 25 & AVAL < 30,    "Overweight",        3,
+  "BMI",    AVAL >= 30 & AVAL < 35,    "Obesity class I",   4,
+  "BMI",    AVAL >= 35 & AVAL < 40,    "Obesity class II",  5,
+  "BMI",    AVAL >= 40,                "Obesity class III", 6,
+  "BMI",    is.na(AVAL),               NA_character_,       NA_integer_
+)
+
+# Read in data ----
+# See the "Read in Data" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#readdata)
+
+# Read data
+vs_metabolic <- admiralmetabolic::vs_metabolic
+adsl <- admiral::admiral_adsl
+
+# Convert SAS empty values to NA
+advs <- vs_metabolic |>
+  convert_blanks_to_na()
+adsl <- adsl |>
+  convert_blanks_to_na()
+
+# Merge ADSL variables (TRTSDT, TRTEDT, TRT01P, TRT01A) needed for ADVS
+# derivations
+advs <- advs |>
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = exprs(TRTSDT, TRTEDT, TRT01P, TRT01A),
+    by_vars = exprs(STUDYID, USUBJID)
```

Comment:
I actually strongly disagree here. If we assume that people will use these templates as something to base their own work on, extracting argument data from an package specific metadata system is a bad idea. 

I can see that this makes a lot of sense inside R code, but think that we should strongly consider if this is a good idea for a template.

Reply:
@AndersAskeland I see your point. It's a tough thing to balance because we _have_ these options and so we should:
1.  let the users be aware they exist, 
2.  use them ourselves in templates.

In general I agree with your concerns though- in fact within admiralroche we have the following code at the top of our Roche-specific templates:
```
# Set subject keys ----
# user to set subject key variables in one place

set_admiral_options(subject_keys = exprs(STUDYID, USUBJID))
```
We could do the same in our templates?